nuget/Microsoft.Windows.CsWinRT.targets

```diff
@@ -6,19 +6,76 @@ Copyright (C) Microsoft Corporation. All rights reserved.
 <Project ToolsVersion="14.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
 
   <PropertyGroup>
-    <GeneratedFilesDir Condition="'$(GeneratedFilesDir)'==''">$([MSBuild]::NormalizeDirectory('$(MSBuildProjectDirectory)', '$(IntermediateOutputPath)', 'Generated Files'))</GeneratedFilesDir>
-    <ResolveAssemblyReferencesDependsOn Condition="'$(RemoveWindowsReference)'=='true'">$(ResolveAssemblyReferencesDependsOn);RemoveWindowsReference</ResolveAssemblyReferencesDependsOn>
-    <AllowUnsafeBlocks Condition="$(CsWinRTComponent)==true">true</AllowUnsafeBlocks>
+    <CsWinRTMessageImportance Condition="'$(CsWinRTMessageImportance)' == ''">normal</CsWinRTMessageImportance>
+    <CsWinRTCommandVerbosity Condition="'$(CsWinRTMessageImportance)' == 'high'">-verbose</CsWinRTCommandVerbosity>
+    <ResolveAssemblyReferencesDependsOn Condition="'$(CsWinRTRemoveWindowsReference)'!='false'">$(CsWinRTRemoveWindowsReference);CsWinRTRemoveWindowsReference</ResolveAssemblyReferencesDependsOn>
+    <CsWinRTEnabled Condition="'$(CsWinRTFilters)$(CsWinRTParams)'==''">false</CsWinRTEnabled>
+    <CsWinRTEnabled Condition="'$(CsWinRTEnabled)'==''">true</CsWinRTEnabled>
+    <CsWinRTWindowsSdkVersion Condition="'$(CsWinRTWindowsSdkVersion)'==''">$(WindowsSDKVersion.TrimEnd('\'))</CsWinRTWindowsSdkVersion>
+    <CsWinRTGenerateProjection Condition="'$(CsWinRTEnabled)'=='false'">false</CsWinRTGenerateProjection>
+    <CsWinRTGenerateProjection Condition="'$(CsWinRTGenerateProjection)'==''">true</CsWinRTGenerateProjection>
+    <AllowUnsafeBlocks Condition="'$(CsWinRTComponent)' == 'true'">true</AllowUnsafeBlocks>
+    <AllowUnsafeBlocks Condition="'$(CsWinRTEnabled)' == 'true'">true</AllowUnsafeBlocks>
   </PropertyGroup>
   
   <!--Remove Windows.Winmd reference to prevent compile collisions-->
-  <Target Name="RemoveWindowsReference" Outputs="@(Reference)">
+  <Target Name="CsWinRTRemoveWindowsReference" Outputs="@(Reference)" Condition="'$(CsWinRTEnabled)' == 'true'">
     <ItemGroup>
       <Reference Remove="Windows"/>
     </ItemGroup>
   </Target>
 
-  <ItemGroup Condition="$(CsWinRTComponent)==true">
+  <Target Name="CsWinRTRemoveWinMDReferences" BeforeTargets="ResolveReferences;CoreCompile" AfterTargets="AfterResolveReferences">
+    <ItemGroup>
+      <!--Move winmd references into private item group to prevent subsequent winmd reference errors-->
+      <CsWinRTRemovedReferences Include="@(ReferencePath)" Condition="'%(ReferencePath.WinMDFile)' == 'true'" />
+      <CsWinRTInputs Include="@(CsWinRTRemovedReferences)"/>
+
+      <!--Prevent NETSDK1130 errors from winmd references-->
+      <ReferencePath Remove="@(ReferencePath)" Condition="'%(ReferencePath.WinMDFile)' == 'true'" />
+      <ReferencePathWithRefAssemblies Remove="@(CsWinRTRemovedReferences)" Condition="%(CsWinRTRemovedReferences.Filename) == '%(ReferencePathWithRefAssemblies.Filename)'" />
```

Comment:
Does FileName include extension?  If not, should we also check extension so that we only remove winmd and not dlls or so that may have same name.

Reply:
CsWinRTRemovedReferences is only collecting winmd files to begin with, so only they would be removed